### PR TITLE
Add traffic generator image param

### PR DESCRIPTION
**Pull request by RamLavi**
This PR adds a new traffic generator image param.
This param is optional and is added in order to allow flexibility in
choosing which traffic generator image to run.

### CODE REVIEW

**Comment by RamLavi** [comment]
checking with premade traffic generator image `quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:initial`

```
zeus09:kubevirt-dpdk-checkup $ TRAFFIC_GEN_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:initial TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           -e RUNTIME_CLASS_NAME= \
           -e TRAFFIC_GEN_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:initial \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1675246408

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:61
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [110.742 seconds]
------------------------------

Ran 1 of 1 Specs in 110.747 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (110.75s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests110.776s
```

File: pkg/internal/checkup/checkup.go [hunk 1]
```diff
@@ -300,7 +300,7 @@ func newTrafficGeneratorPod(checkupConfig config.Config, secondaryNetworkRequest
 		[]k8scorev1.Capability{"IPC_LOCK", "SYS_RESOURCE", "NET_RAW", "NET_ADMIN"})
 
 	trafficGeneratorContainer := pod.NewPodContainer(TrafficGeneratorPodNamePrefix,
-		pod.WithContainerImage(pod.ContainerImage),
```
Comment (orelmisan) [inline pkg/internal/checkup/checkup.go:303]: I don't see a removal of this constant.
Where else is it used?

File: tests/test_suite_test.go [hunk 2]
```diff
@@ -79,4 +81,6 @@ var _ = BeforeSuite(func() {
 	if runtimeClassName = os.Getenv(runtimeClassNameVarName); runtimeClassName == "" {
 		runtimeClassName = defaultRuntimeClassName
 	}
+
+	trafficGeneratorImage = os.Getenv(trafficGeneratorImageVarName)
```
Comment (orelmisan) [inline tests/test_suite_test.go:85]: Why is this environment variable is treated differently than the others?

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi.
Could you please add more details about this change in the PR description?

File: tests/test_suite_test.go (hunk 2 above)
Comment (RamLavi) [reply tests/test_suite_test.go:85]: I don't think it should have a default value set here, because we already do that on the config package...

**Comment by RamLavi** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/ba010f74363f2218e5bea8bdbafcfbe921b9004f..9cc2510fc557f2b7da5c030e6bcab7a3e31e3b7c): review fixes

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TRAFFIC_GEN_IMAGE_URL=quay.io/schseba/trex:2.87 TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           -e RUNTIME_CLASS_NAME= \
           -e TRAFFIC_GEN_IMAGE_URL=quay.io/schseba/trex:2.87 \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1675253764

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:61
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [100.796 seconds]
------------------------------

Ran 1 of 1 Specs in 100.800 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (100.80s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	100.833s
```

**Review by orelmisan — APPROVED** [review]

**Pull request merged**